src/main/java/com/epam/javacourses2016/task17/SolverTask17.java

```diff
@@ -18,8 +21,76 @@
      * @param segments Множество отрезков.
      * @return Множество точек пересечения, имеющих минимальную абсциссу.
      */
-    Set<Point2D> analyze(Set<Segment> segments) {
-        //TODO
-        return null;
+    public Set<Point2D> analyze(Set<Segment> segments) {
+        if (segments == null) {
+            return null;
+        }
+        if (segments.size() <= 1) {
+            return null;
+        }
+
+        List<Segment> listSegments = new ArrayList<>(segments);
+        Set<Point2D> resultSet = new HashSet<>();
+        double minX = Double.MAX_VALUE;
+
+        for (int i = 0; i < segments.size(); i++) {
```

Comment:
I added sort now. Also added some breaks from iteration. Also thought about TreeSet here, because i dont rly need map value here(or i missing somethin).